Updated app to use new site config structure

gsidebo opened the pull request:
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #120 

#### What's this PR do?
- Adds a management command to override site configs in the local db
- Updates the front and back end to use the new config format (see issue for details), particularly in the site sidebar where content editing links are generated

#### How should this be manually tested?
- Use the new management command to update the local site config for the course site starter
- Modify the override config to add/remove new menu items, then make sure the sidebar looks correct and links to the correct pages with the correct forms

#### Any background context you want to provide?
- I added the YAML file and management command for overriding site configs in the db because the practice of making changes to the course site config, then creating data migrations to adapt the config JSON to match it was not sustainable. We should drop that practice and apply course site config changes in that override YAML file.
- I removed the page/resource constants because this new config is designed to be more generalized/arbitrary. We no longer have a need to limit WebsiteContent records to be either a page or a resource.

#### Screenshots (if appropriate)
![ss 2021-03-18 at 15 03 57 ](https://user-images.githubusercontent.com/14932219/111682832-33c4ab00-87fb-11eb-87b9-3dbb2ea0e2c9.png)

noisecapella commented on static/js/components/SiteSidebar.tsx:
```diff
@@ -1,45 +1,99 @@
+/*
+ * This function adds a "Settings" menu header if it doesn't exist, then adds a link to the Collaborators page
+ */
+const addCollaboratorMenuItem = (
```
It looks like this function and the `forEach` below do very similar things. Maybe this should be a utility function:

    const addToList = (map: Map<string, Array<NavMenuItem>, key: string, value: NavMenuItem) => {
        const list = map.get(key) ?? []
        map.set(key, [...list, value])
    }

And then below:

    const configMap = new Map<string, Array<ConfigItem>>()
    configItems.forEach((configItem: ConfigItem) => {
      addToList(configMap, configItem.category, configItem)
    })
    if (website.is_admin) {
      addToList(configMap, "Settings", collaboratorMenuItem)
    }

What do you think?
    }

noisecapella commented on static/js/components/forms/SiteAddContentForm.test.tsx:
```diff
@@ -2,39 +2,52 @@ import React from "react"
     sandbox = sinon.createSandbox()
     setFieldValueStub = sinon.stub()
-    website = makeWebsiteDetail()
     // @ts-ignore
```
We can remove this line now, it was only there for the `find(...)` function

noisecapella commented on static/js/components/forms/SiteEditForm.test.tsx:
```diff
@@ -5,32 +5,28 @@ import { shallow } from "enzyme"
     setFieldValueStub = sinon.stub()
-    website = makeWebsiteDetail()
     content = makeWebsiteContentDetail()
     // @ts-ignore
```
We can remove this `@ts-ignore`

gsidebo commented:
@noisecapella Ready for review again, and your comment reminded me that we should probably rename `SiteEditForm` if we renamed `SiteAddForm`. I added that to the commit as well
